bitte/pab.nix

```diff
@@ -79,31 +70,33 @@ let
     set -eEuo pipefail
 
     echo "[pab-init-cmd]: Dropping PAB database file '${dbFile}'" >&2
-    rm -rf ${dbFile}
+    rm -rf "${dbFile}"
 
     echo "[pab-init-cmd]: Creating new DB '${dbFile}'" >&2
-    ${pabExe} --config=${pabYaml} migrate;
+    ${pabExe} --config=pab.yaml migrate
   '';
 in
 writeShellScriptBin "entrypoint" ''
   set -eEuo pipefail
 
-  export PATH=${lib.makeBinPath [ coreutils gnused utillinux ]}
+  export PATH=${lib.makeBinPath [ coreutils gnused utillinux wait-for-socket ]}
 
   export SYSTEM_CERTIFICATE_PATH=${cacert}/etc/ssl/certs/ca-bundle.crt
 
-  sed -e "s|@WEBSERVER_PORT@|$((PORT_RANGE_BASE))|g" \
-      -e "s|@NODE_PORT@|$((PORT_RANGE_BASE + 1))|g" \
-      -e "s|@CHAIN_INDEX_PORT@|$((PORT_RANGE_BASE + 2))|g" \
-      -e "s|@SIGNING_PROCESS_PORT@|$((PORT_RANGE_BASE + 3))|g" \
-      -e "s|@WALLET_PORT@|$((PORT_RANGE_BASE + 4))|g" \
-      ${pabYamlIn} > ${pabYaml}
+  sed -e "s|@PAB_STATE_DIR@|$PAB_STATE_DIR|g" \
+      -e "s|@NOMAD_PORT_pab@|$NOMAD_PORT_pab|g" \
+      -e "s|@NOMAD_ADDR_wbe@|$NOMAD_ADDR_wbe|g" \
+      -e "s|@NOMAD_ADDR_node@|$NOMAD_ADDR_node|g" \
+      -e "s|@NOMAD_ALLOC_DIR@|$NOMAD_ALLOC_DIR|g" \
+      -e "s|@NOMAD_ADDR_index@|$NOMAD_ADDR_index|g" \
+      ${pabYamlIn} > pab.yaml
 
+  wait-for-socket "$NOMAD_ALLOC_DIR/node.sock"
 
   ${pab-init-cmd}/bin/pab-init-cmd
 
   # Ugly ugly hack to kill the PAB at midnight UTC
-  ${pabExe} --config=${pabYaml} all-servers&
+  ${pabExe} --config=pab.yaml webserver &
```

Comment:
Is midnight UTC the best time for this? I'm wondering what the UTC time is when this would least impact users?

Reply:
This is what was decided for the summit earlier this year. I don't know currently how our users our distributed... Ideally we'd just fix the bug causing memory leak over time though.